src/ows/src/main/java/org/geoserver/ows/FilePublisher.java

```diff
@@ -61,12 +70,32 @@ protected void initServletContext(ServletContext servletContext) {
     }
 
     @Override
-    protected URL getUrl(HttpServletRequest request) throws IOException {
-        String ctxPath = request.getContextPath();
-        String reqPath = request.getRequestURI();
-        reqPath = URLDecoder.decode(reqPath, "UTF-8");
-        reqPath = reqPath.substring(ctxPath.length());
+    protected String getMimeType(String reqPath, String filename) {
+        String mimeType = super.getMimeType(reqPath, filename);
+        if (!"/index.html".equals(reqPath)) {
+            String lowerCaseMime = mimeType.toLowerCase();
+            // force using the static web files directory for html/javascript files and only allow
+            // the html/javascript mime types when the system property is set to true to mitigate
+            // stored XSS vulnerabilities
+            if (lowerCaseMime.contains("html") || lowerCaseMime.contains("javascript")) {
+                String prop = GeoServerExtensions.getProperty(DISABLE_STATIC_WEB_FILES);
```

Comment:
It may be worth looking this value up on construction; although this is not very performance sensitive.

Reply:
I don't think that would be too much of a concern here. At worst, GeoServerExtensions.getProperty is basically just doing a lookup in four different maps which is not a big deal. Also, it is known at this point that there is a valid HTML/JS file that will need to be read from the file system and transmitted to the user so calling getProperty is just a tiny fraction of the request runtime.